src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c

```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
     (ActorFunc)EnHorseLinkChild_Update,
     (ActorFunc)EnHorseLinkChild_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A693D0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A6948C.s")
+static AnimationHeader* sAnimations[] = { 0x060043E4, 0x06004B08, 0x060053F0, 0x0600360C, 0x06002F98 };
+static ColliderCylinderInit_Set3 sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 20, 100, 0, { 0, 0, 0 } }
+};
+static ColliderJntSphItemInit sJntSphItemInit[] = {
+    { { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+      { 0xD, { { 0, 0, 0 }, 10 }, 100 } }
+};
+static ColliderJntSphInit sJntSphInit = { { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x12, COLSHAPE_JNTSPH },
+                                          ARRAY_COUNT(sJntSphItemInit),
+                                          sJntSphItemInit };
+static CollisionCheckInfoInit sColCheckInfoInit = { 10, 35, 100, 0xFE };
+
+extern AnimationHeader D_06002F98;
+extern SkeletonHeader D_06007B20;
+
+void func_80A693D0(EnHorseLinkChild* this) {
+    static s32 D_80A6AF5C[] = { 1, 19 };
+
+    if ((D_80A6AF5C[this->unk_1F0] < this->skin.skelAnime.animCurrentFrame) &&
+        !((this->unk_1F0 == 0) && (D_80A6AF5C[1] < this->skin.skelAnime.animCurrentFrame))) {
```

Comment:
if it matches, I would put the array accesses on the other side of the comparison
`if (frame count is > array accesses)`